### PR TITLE
Remove unnecessary NVF_API from scheduler/utils.h

### PR DESCRIPTION
**Pull request by wujingyue**

### CODE REVIEW

**Comment by wujingyue** [comment]
!test

**Review by tfogal — APPROVED** [review]
Thanks!

Please do LD_PRELOAD such a library into thunder and run e.g. test_nvfuser.py there or something, to double-check this is fine.

**Comment by wujingyue** [comment]
!test

**Comment by wujingyue** [comment]
> Please do LD_PRELOAD such a library into thunder and run e.g. test_nvfuser.py there or something, to double-check this is fine.

Do you mean the following? I failed to understand how LD_PRELOAD helps to verify. 

```
$ LD_PRELOAD=/opt/pytorch/nvfuser/bin/libnvfuser.so pytest thunder/tests/test_nvfuser.py
================================================================================================ test session starts =================================================================================================
platform linux -- Python 3.12.3, pytest-8.1.1, pluggy-1.5.0
Test order randomisation NOT enabled. Enable with --random-order or --random-order-bucket=<bucket_type>
benchmark: 5.1.0 (defaults: timer=torch.utils.benchmark.utils.timer.timer disable_gc=False min_rounds=5 min_time=0.000005 max_time=1.0 calibration_precision=10 warmup=True warmup_iterations=100000)
rootdir: /opt/pytorch/lightning-thunder
configfile: pyproject.toml
plugins: xdist-3.6.1, timestamper-0.0.10, cov-6.0.0, timeout-2.3.1, random-order-1.1.1, hypothesis-6.122.3, mpi-0.6, benchmark-5.1.0, shard-0.1.2, typeguard-4.3.0
timeout: 900.0s
timeout method: signal
timeout func_only: False
collected 47 items
Running 47 items in this shard

thunder/tests/test_nvfuser.py ..............................ssssssssssssssss.                                                                                                                                  [100%]

=================================================================================== 31 passed, 16 skipped, 233 warnings in 11.22s ====================================================================================
```

**Comment by tfogal** [comment]
> > Please do LD_PRELOAD such a library into thunder and run e.g. test_nvfuser.py there or something, to double-check this is fine.
> 
> Do you mean the following? I failed to understand how LD_PRELOAD helps to verify.
> 
> ```
> $ LD_PRELOAD=/opt/pytorch/nvfuser/bin/libnvfuser.so pytest thunder/tests/test_nvfuser.py
> ...
> ```

I did, yeah that works.
While initially adding that support, there were some issues of hidden symbols that somehow were not tickled by our tests.

Side note, is there a way to run thunder's tests on an nvFuser patch like this via the `!test` comments? @xwang233 ? Low priority.

**Comment by tfogal** [comment]
> We do run thunder tests on Nvfuser PR's. Do you mean to add an additional thunder test with `LD_PRELOAD=libnvfuser.so` ?

Probably not. One should only need the `LD_PRELOAD` hack if there are multiple nvFusers installed on a system.
